ksqldb-engine/src/main/java/io/confluent/ksql/services/KafkaConsumerGroupClientImpl.java

```diff
@@ -52,7 +56,10 @@ public ConsumerGroupSummary describeConsumerGroup(final String group) {
     try {
       final Map<String, ConsumerGroupDescription> groups = ExecutorUtil
           .executeWithRetries(
-              () -> adminClient.describeConsumerGroups(Collections.singleton(group)).all().get(),
+              () -> adminClient.get()
```

Comment:
In this case I followed `KafkaTopicClient` design as it is the most similar in terms of behavior and instance creatin.

AFAIK, as well as Consumer and Producer, Admin client does not do any handshake when instantiated; though it make sense to reduce unnecessary recreations.

Following creation path, `KafkaConsumerGroupClient` and `KafkaTopicClient` both receive admin client supplier from `ServiceContext` which has all instances `memoized` to avoid recreation, e.g.:
`private final MemoizedSupplier<Admin> adminClientSupplier;`

With this design I don't see a need to change the current behavior. wdyt?

Reply:
Yeah, this follows the same patterns as `KafkaTopicClient`.  The Memorized supplier pattern is used to _avoid_ Admin Clients being instantiated, potentially per request due to different user credentials, during service context creation.  Admin client instantiation is expensive and should be done lazily, as is the case here.